add grace period for evacuate

pweil- opened the pull request:
Fixes https://github.com/openshift/origin/issues/4911

@fabianofranz @deads2k @derekwaynecarr @pmorie @eparis @smarterclayton

smarterclayton commented on pkg/cmd/admin/node/evacuate.go:
```diff
@@ -25,6 +26,8 @@ func (e *EvacuateOptions) AddFlags(cmd *cobra.Command) {
 
 	flags.BoolVar(&e.DryRun, "dry-run", false, "Show pods that will be migrated. Optional param for --evacuate")
 	flags.BoolVar(&e.Force, "force", false, "Delete pods not backed by replication controller. Optional param for --evacuate")
+	flags.Int64Var(&e.Grace, "grace", 30, "Grace period (seconds) for pods being deleted. Optional param for --evacuate")
```
Should be identical to upstream, i.e. `grace-period`

pweil- replied:
updated name and vars

smarterclayton commented:
Test case please.

pweil- commented:
> Test case please.

Added test that expected flags are registered and have expected default values.  Refactored making delete options to its own method so I can test that it is created correctly.  We could probably take this further by refactoring NodeOptions into an interface so commands could mock the behavior and test the expected calls are created.  Perhaps that's why other commands in the package don't have tests at the moment.

PTAL

deads2k commented on pkg/cmd/admin/node/evacuate.go:
```diff
@@ -12,19 +12,32 @@ import (
 
-	flags.BoolVar(&e.DryRun, "dry-run", false, "Show pods that will be migrated. Optional param for --evacuate")
-	flags.BoolVar(&e.Force, "force", false, "Delete pods not backed by replication controller. Optional param for --evacuate")
+	flags.BoolVar(&e.DryRun, flagDryRun, defaultDryRun, "Show pods that will be migrated. Optional param for --evacuate")
```
Using the current value of `e.DryRun` makes it possible for different composers to choose different defaults.

pweil- replied:
added New* method in the same pattern as most other commands and changed the manage node command to use it.  Updated tests.

deads2k commented:
Optional nit.  lgtm.

smarterclayton commented:
[merge]
